src/test/java/redis/clients/jedis/csc/ClientSideCacheFunctionalityTest.java

```diff
@@ -21,56 +32,164 @@
 
 public class ClientSideCacheFunctionalityTest extends ClientSideCacheTestBase {
 
-  @Test
-  public void flushEntireCache() {
-    int count = 100;
+  @Test // T.5.1
+  public void flushAllTest() {
+    final int count = 100;
     for (int i = 0; i < count; i++) {
       control.set("k" + i, "v" + i);
     }
 
-    HashMap<CacheKey, CacheEntry> map = new HashMap<>();
-    Cache clientSideCache = new TestCache(map);
-    JedisPooled jedis = new JedisPooled(hnp, clientConfig.get(), clientSideCache);
-    try {
+    Cache cache = new TestCache();
+    try (JedisPooled jedis = new JedisPooled(hnp, clientConfig.get(), cache)) {
       for (int i = 0; i < count; i++) {
         jedis.get("k" + i);
       }
 
+      assertEquals(count, cache.getSize());
+      cache.flush();
+      assertEquals(0, cache.getSize());
+    }
+  }
+
+  @Test // T.4.1
+  public void lruEvictionTest() {
+    final int count = 100;
+    final int extra = 10;
+
+    // Add 100 + 10 keys to Redis (e.g., SET key:1 ... SET key:100)
+    for (int i = 0; i < count + extra; i++) {
+      control.set("key:" + i, "value" + i);
+    }
+
+    Map<CacheKey, CacheEntry> map = new LinkedHashMap<>(count);
+    Cache cache = new DefaultCache(count, map);
+    try (JedisPooled jedis = new JedisPooled(hnp, clientConfig.get(), cache)) {
+
+      // Retrieve the 100 keys in the same order (e.g., GET key:1 ... GET key:100)
+      for (int i = 0; i < count; i++) {
+        jedis.get("key:" + i);
+      }
+      assertThat(map, Matchers.aMapWithSize(count));
+
+      List<CacheKey> earlierKeys = new ArrayList<>(map.keySet()).subList(0, extra);
+      // earlier keys in map
+      earlierKeys.forEach(cacheKey -> assertThat(map, Matchers.hasKey(cacheKey)));
+
+      // Retrieve the 10 extra keys
+      for (int i = count; i < count + extra; i++) {
+        jedis.get("key:" + i);
+      }
+
+      // earlier keys NOT in map
+      earlierKeys.forEach(cacheKey -> assertThat(map, Matchers.not(Matchers.hasKey(cacheKey))));
+      assertThat(map, Matchers.aMapWithSize(count));
+    }
+  }
+
+  @Test // T.5.2
+  public void deleteByKeyTest() {
+    final int count = 100;
+    for (int i = 0; i < count; i++) {
+      control.set("k" + i, "v" + i);
+    }
+
+    // By using LinkedHashMap, we can get the hashes (map keys) at the same order of the actual keys.
+    LinkedHashMap<CacheKey, CacheEntry> map = new LinkedHashMap<>();
+    Cache clientSideCache = new TestCache(map);
+    try (JedisPooled jedis = new JedisPooled(hnp, clientConfig.get(), clientSideCache)) {
+      for (int i = 0; i < count; i++) {
+        jedis.get("k" + i);
+      }
       assertEquals(count, map.size());
-      clientSideCache.flush();
+
+      ArrayList<CacheKey> cacheKeys = new ArrayList<>(map.keySet());
+      for (int i = 0; i < count; i++) {
```

Comment:
nit: I think the scenario does not require you to delete more keys than one

Reply:
Added the test above this one.